### ISSUE TITLE
loss tend to be nan or inf

### ISSUE THREAD

**Issue by BaohaoLiao** [yxli2123/LoftQ#4]
Thank you for this great work!

May I ask whether the reported results are from fake_quantization=True or fake_quantization=False? When I use fake_quantization=False, the training doesn't succeed for the gsm8k task for llama-2-7b, always leading to nan or inf loss.

**Comment by BaohaoLiao** [comment]
Resolved. It is caused by the cuda memory. Since my A100 only has 40GB, I need to decrease the per_device_batch_size

**Comment by PuNeal** [comment]
@yxli2123 Hi, Thanks for your great work. I find that loftq has been merged into peft in [https://github.com/huggingface/peft/pull/1150](url), then I run examples/loftq_finetuning with commands in [https://github.com/huggingface/peft/blob/main/examples/loftq_finetuning/README.md](url):
```sh
python train_gsm8k_llama.py \
    --model_name_or_path LoftQ/Llama-2-7b-hf-4bit-64rank \
    --output_dir exp_results/gsm8k/llama-2-7b/bit4-rank64/lr3e-4 \
    --learning_rate 3e-4  \
    --seed 202 \
    --dataset_name gsm8k \
    --dataset_config main \
    --pad_to_max_length \
    --max_source_length 128 \
    --max_target_length 256 \
    --num_train_epochs 5 \
    --per_device_train_batch_size 4 \
    --per_device_eval_batch_size 4 \
    --gradient_accumulation_steps 4 \
    --with_tracking \
    --report_to tensorboard
```
But I also met this problem: loss tend to be nan after ~1200 steps.

**Comment by yxli2123** [comment]
Hi @PuNeal, I have rewritten the training script. Please try again.

**Comment by BaohaoLiao** [comment]
I didn't meet this problem again with the new update
